belt_nlp/bert_embedder_pooling.py

```diff
@@ -0,0 +1,112 @@
+import torch
+
+from belt_nlp.bert_with_pooling import BertClassifier
+from typing import Optional, List
+from torch import Tensor
+from torch.nn import Module
+from transformers import PreTrainedTokenizerBase, BatchEncoding
+from belt_nlp.splitting import transform_list_of_texts
+
+
+class BertEmbeddingGenerator(BertClassifier):
```

Comment:
This inheritance from BertClassifier is confusing - this is not the classifier. I suggest refactoring it using [composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance) principle.